index.bs

```diff
@@ -1840,29 +1806,125 @@ Each [=mock sensor type=] has a [=mock sensor reading values=] dictionary:
+    1.  If |topLevelVirtualSensorMapping| does not [=map/contain=] |sensorType|, return [=error=] with
+        [=WebDriver error code=] [=invalid argument=].
+    1.  Let |virtualSensor| be |topLevelVirtualSensorMapping|[|sensorType|].
+    1.  Let |parsedReading| be the result of invoking |metadata|'s [=virtual sensor metadata/reading
+        parsing algorithm=] with |reading|.
+    1.  If |parsedReading| is **undefined**, return [=error=] with [=WebDriver error code=]
+        [=invalid argument=].
+    1.  In an [=implementation-defined=] way, make |parsedReading| available so that it can be
+        obtained by [=platform sensors=] connected to |virtualSensor|.
     1.  Return [=success=] with data `null`.
 </div>
 
-### Delete mock sensor ### {#delete-mock-sensor-command}
+#### Algorithms for parsing readings #### {#algorithms-for-parsing-readings}
+
+This specification defines some algorithms that [=extension specifications=] can use when defining
+a [=virtual sensor metadata=] for use in [=per-type virtual sensor metadata=].
+
+<h6 dfn export>Parse single-value number reading</h6>
 
-<table>
+<div algorithm="parse single-value number reading">
+    : input
+    :: |parameters|, a JSON {{Object}}
+    :: |valueName|, a [=string=]
+    : output
+    :: A [=sensor reading=] or **undefined**
+
+    1. Let |value| be the result of invoking [=get a property=] from |parameters| with |valueName|.
+    1. If |value| is not a {{Number}}, or its value is **NaN**, +∞, or −∞, return **undefined**.
+    1. Let |reading| be a new [=sensor reading=].
+    1. [=map/Set=] |reading|[|valueName|] to |value|.
+    1. Return |reading|.
+</div>
+
+<h6 dfn export>Parse XYZ reading</h6>
+
+<div algorithm="parse xyz reading">
+    : input
+    :: |parameters|, a JSON {{Object}}
+    : output
+    :: A [=sensor reading=] or **undefined**
+
+    1. Let |x| be the result of invoking [=get a property=] from |parameters| with "`x`".
+    1. If |x| is not a {{Number}}, or its value is **NaN**, +∞, or −∞, return **undefined**.
+    1. Let |y| be the result of invoking [=get a property=] from |parameters| with "`y`".
+    1. If |y| is not a {{Number}}, or its value is **NaN**, +∞, or −∞, return **undefined**.
+    1. Let |z| be the result of invoking [=get a property=] from |parameters| with "`z`".
+    1. If |z| is not a {{Number}}, or its value is **NaN**, +∞, or −∞, return **undefined**.
+    1. Let |reading| be a new [=sensor reading=].
+    1. [=map/Set=] |reading|["`x`"] to |x|.
+    1. [=map/Set=] |reading|["`y`"] to |y|.
+    1. [=map/Set=] |reading|["`z`"] to |z|.
+    1. Return |reading|.
+</div>
+
+<h6 dfn export>Parse quaternion reading</h6>
+
+<div algorithm="parse quaternion reading">
+    : input
+    :: |parameters|, a JSON {{Object}}
+    : output
+    :: A [=sensor reading=] or **undefined**
+
+    1. Let |quaternionArray| be the result of invoking [=get a property=] from |parameters| with
+       "`quaternion`".
```

Comment:
I guess it is consistent with the `OrientationSensor` interface which puts the value into a `FrozenArray<double>` rather than having x, y, z and w properties like `Accelerometer` has.